Common++/src/IpAddress.cpp

```diff
@@ -166,33 +140,29 @@ namespace pcpp
 	}
 
 
-	bool IPv6Address::matchSubnet(const IPv6Address& subnet, uint8_t prefixLength) const
-	{
-		try
-		{
-			auto ipv6Network = IPv6Network(subnet, prefixLength);
-			return ipv6Network.includes(*this);
-		}
-		catch (const std::invalid_argument& e)
-		{
-			PCPP_LOG_ERROR(e.what());
-			return false;
-		}
-	}
-
-
 	// ~~~~~~~~~
 	// IPAddress
 	// ~~~~~~~~~
 
 
-	IPAddress::IPAddress(const std::string& addrAsString) : m_Type(IPv6AddressType), m_IPv6(addrAsString)
+	IPAddress::IPAddress(const std::string& addrAsString)
```

Comment:
I refactored the logic of the construction of `IPAddress`

Reply:
Using `inet_pton` is an internal implementation detail of `IPv4Address` / `IPv6Address`. I think it'd be nicer to either:
- Try to create `IPv4Address`, if it throws an exception try to create `IPv6Address`
- Or add a static `tryParse()` method to both `IPv4Address` and `IPv6Address` that try to create an object without throwing an exception

Reply:
@seladb  we didn't expose the implementation detail of `IPAddress::IPAddress`. I don't see the drawback using `inet_pton` to check if it is IPv4 or IPv6? 

I think we shouldn't use exception to test if it is IPv4 or not, and even if we provide `tryParse()`, the new function will still need to use `inet_pton` underlying.

Reply:
I think it's nicer to keep `inet_pton` inside `IPv4Address` / `IPv6Address` rather than exposing it to `IPAddress`. It's a better separation of concerns to let a class decide if its data is valid or not, and not let another class determine for it

Reply:
sure, let me refactor this part of code.

Reply:
sorry, forgot this one.  fixed.